solr/core/src/java/org/apache/solr/handler/replication/BackupAPI.java

```diff
@@ -0,0 +1,100 @@
+package org.apache.solr.handler.replication;
+
+import static org.apache.solr.handler.ClusterAPI.wrapParams;
+import static org.apache.solr.security.PermissionNameProvider.Name.CORE_EDIT_PERM;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.parameters.RequestBody;
+import java.util.HashMap;
+import java.util.Map;
+import javax.inject.Inject;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import org.apache.solr.api.JerseyResource;
+import org.apache.solr.common.annotation.JsonProperty;
+import org.apache.solr.core.CoreContainer;
+import org.apache.solr.handler.ReplicationHandler;
+import org.apache.solr.jersey.JacksonReflectMapWriter;
+import org.apache.solr.jersey.PermissionName;
+import org.apache.solr.jersey.SolrJerseyResponse;
+import org.apache.solr.request.SolrQueryRequest;
+import org.apache.solr.response.SolrQueryResponse;
+
+/** V2 endpoint for Backup API used for User-Managed clusters and Single-Node Installation. */
+@Path("/cores/{cores}/replication/backups")
+public class BackupAPI extends JerseyResource {
```

Comment:
[-1] Because Solr has a bunch of different "backup"-related features and APIs, I think we should come up with a more descriptive name here.  Maybe "ReplicationBackupAPI", since that's where the functionality initially existed?  Maybe "SnapshotBackupAPI", since this triggers the "snapshot" functionality and not a "true" backup?  Idk.  Any ideas?

Reply:
I have created new package for it and named it 'replication' under 'handler' package. And I was thinking of moving all the replication related commands under this package.
But if we do not want new package then Yes I can change it's name to suggested name and move it to /handler/admin/api

Reply:
I didn't notice the package change.  I don't introduce new packages too often, mostly because Solr is very disorganized and inconsistent in its use of them.  Though my attitude there is probably part of the problem. 

Makes sense to me; forget I said anything.